Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/ExotelEndRide.hs

```diff
@@ -34,7 +37,24 @@ import Tools.Error
 
 type AckResp = AckResponse
 
-callBasedEndRide :: (EsqDBFlow m r, CacheFlow m r, EsqDBReplicaFlow m r, HasField "enableAPILatencyLogging" r Bool, HasField "enableAPIPrometheusMetricLogging" r Bool) => EndRide.ServiceHandle m -> Id Merchant -> DbHash -> Text -> m AckResp
+callBasedEndRide ::
+  ( EsqDBFlow m r,
+    CacheFlow m r,
+    EsqDBReplicaFlow m r,
+    HasField "enableAPILatencyLogging" r Bool,
+    HasField "enableAPIPrometheusMetricLogging" r Bool,
+    HasFlowEnv m r '["smsCfg" ::: SmsConfig],
+    HasCacheConfig r,
+    EsqDBFlow m r,
+    EsqLocDBFlow m r,
+    EncFlow m r,
+    Redis.HedisFlow m r
```

Comment:
`EsqDBFlow m r` duplicated
`HasCacheConfig r`, `Redis.HedisFlow m r` redundant

Reply:
removed the duplicated and redundant constraints